core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -203,7 +205,8 @@ private[spark] class ExecutorAllocationManager(
       throw new SparkException(
         s"s${DYN_ALLOCATION_SUSTAINED_SCHEDULER_BACKLOG_TIMEOUT.key} must be > 0!")
     }
-    if (!conf.get(config.SHUFFLE_SERVICE_ENABLED)) {
+    if (!conf.get(config.SHUFFLE_SERVICE_ENABLED) &&
+      !shuffleDriverComponents.supportsReliableStorage()) {
```

Comment:
Should we not fix the usage of `config.SHUFFLE_SERVICE_ENABLED`? When it is `true`, the code everywhere assumes ESS is enabled. Instead there can be a different config that points to what kind of shuffle service is enabled.

Reply:
`SHUFFLE_SERVICE_ENABLED` will continue to exist for backward compatibility, but imo would be specific for the default implementation for SPARK-25299.

That should be better handled as part of a follow up to SPARK-25299 imo.

Reply:
Ok